.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,27 @@
+name: E2E Shielder tests.
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  rust-checks:
+    runs-on: self-hosted
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
```

Comment:
needed?

Reply:
shouldn't be...

Reply:
See if `toolchain.yaml` with a version won't solve this (next comment)